src/components/input/TokenSelectField.tsx

```diff
@@ -22,6 +21,9 @@ const DEFAULT_VALUE = {
 export function TokenSelectField({ name, label, onChange }: Props) {
   const [field, , helpers] = useField<string>(name)
 
+  const { chain } = useNetwork()
+  const tokenOptions = chain ? getTokenOptionsByChainId(chain?.id) : Object.values(TokenId)
```

Comment:
Let's wrap this in useMemo please

Reply:
> @denviljclarke This mostly looks good, thanks for the contribution! We should merge the two USDC tokens, which should be more clear after you merge latest into your branch. Would you like to fix this or would you prefer I adopt your branch and do the finishing touches? Either way is fine for me :)

Thank you for taking the time to review this! I've made the suggested changes adding useMemo and have added an explanation as to why I introduced a separate token in the list for axlUSDC.

Please feel free to adopt the branch and make any further changes or adjustments you think would improve this. Looking forward to your thoughts!